### PR TITLE
get status- all 3 layers

### PR DESCRIPTION
**Pull request by boazco**

### CODE REVIEW

File: src/main/kotlin/com/hibob/feedback/dao/FeedbackDao.kt [hunk 1]
```diff
@@ -61,4 +61,14 @@ class FeedbackDao(private val sql: DSLContext) {
...
     }
+
+    fun getStatus(feedbackId: UUID, activeUser: ActiveUser): Status {
+        return sql.select(feedbackTables.status)
+            .from(feedbackTables)
+            .where(feedbackTables.feedbackId.eq(feedbackId))
+            .and(feedbackTables.companyId.eq(activeUser.companyId))
+            .and(feedbackTables.employeeId.eq(activeUser.employeeId))
+            .fetchOne()?.let { Status.valueOf(it[feedbackTables.status].toString().uppercase()) }
+            ?: throw BadRequestException("feedback not found")
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/feedback/dao/FeedbackDao.kt]:
```suggestion
            ?: throw NotFoundException("feedback not found")
```

File: src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt [hunk 2]
```diff
@@ -53,4 +53,10 @@ class FeedbackResource(private val feedbackService: FeedbackService) {
         }
     }
 
+    @GET
+    @Path("/v1/getFeedbackStatus/{feedbackId}")
+    fun getStatus(activeUser: ActiveUser, @PathParam("feedbackId") feedbackId: UUID): Response {
+        return Response.ok(feedbackService.getStatus(feedbackId, activeUser)).build()
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt:59]: no permission check?

File: src/main/kotlin/com/hibob/feedback/dao/FeedbackDao.kt [hunk 3]
```diff
@@ -61,4 +61,14 @@ class FeedbackDao(private val sql: DSLContext) {
             .fetchOne(feedbackMapper)
             ?: throw BadRequestException("Feedback not found")
     }
+
+    fun getStatus(feedbackId: UUID, activeUser: ActiveUser): Status {
+        return sql.select(feedbackTables.status)
+            .from(feedbackTables)
+            .where(feedbackTables.feedbackId.eq(feedbackId))
+            .and(feedbackTables.companyId.eq(activeUser.companyId))
+            .and(feedbackTables.employeeId.eq(activeUser.employeeId))
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/feedback/dao/FeedbackDao.kt:70]: the only person who can look at the status of a request is the person who created it?

Comment (boazco) [reply src/main/kotlin/com/hibob/feedback/dao/FeedbackDao.kt:70]: Status Checking API: Provide an API endpoint for employees to check the status of their submitted feedback (if not anonymous).

the spec sounds like that

File: src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt (hunk 2 above)
Comment (boazco) [reply src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt:59]: Status Checking API: Provide an API endpoint for employees to check the status of their submitted feedback (if not anonymous).

the permission check verify that you are HR or Admin but I thought that its for the employee only

File: src/main/kotlin/com/hibob/feedback/dao/FeedbackDao.kt (hunk 3 above)
Comment (boazco) [reply src/main/kotlin/com/hibob/feedback/dao/FeedbackDao.kt:70]: I thought that this API is for employee only and the Filter is for the HR/ admin

File: src/main/kotlin/com/hibob/academy/service/SessionService.kt [hunk 4]
```diff
@@ -35,7 +35,6 @@ class SessionService(private val employeesDao: EmployeesDao) {
         val employee =
             employeesDao.getEmployeeByActiveUser(activeUser) //throws if no employee with this id in this company
         if (employee.role != activeUser.role || employee.department != activeUser.department) {
-            throw BadRequestException("The employee info is not accurate (Role/Department)")
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/service/SessionService.kt:38]: why does this keep showing up in different pr's?

**Review by elijulian — APPROVED** [review]

File: src/main/kotlin/com/hibob/academy/service/SessionService.kt (hunk 4 above)
Comment (boazco) [reply src/main/kotlin/com/hibob/academy/service/SessionService.kt:38]: I think its a mistake that I done in rebase yesterday

Comment (boazco) [reply src/main/kotlin/com/hibob/academy/service/SessionService.kt:38]: I fixed it rn in other small pr and will do rebase for every other branch
